scripts/spec/kickstart_generator_spec.rb

```diff
@@ -0,0 +1,43 @@
+require 'spec_helper'
+require 'kickstart_generator'
+
+describe Build::KickstartGenerator do
+  describe "#run" do
+    def build_base
+      data_file_path("kickstart_generator")
+    end
+
+    def generated
+      "#{build_base}/kickstarts/generated"
+    end
+
+    def ks_text
+      File.read("#{generated}/base.ks")
+    end
+
+    after(:each) do
+      FileUtils.rm_f(Dir.glob("#{generated}/*"))
+    end
+
+    before(:each) do
+      @generator = described_class.new(build_base, ["target"], "puddle", "app_checkout", "miq_checkout")
+      @generator.run(nil)
```

Comment:
This is fine, maybe `run` should return an object you can call GeneratedKickstart#config to return a String of the ini (readable) format and #json to return a json String which is what imagefactory needs to use.

You could then do:

```ruby
expect(@generator.run(nil).config).to include("base file text")
```

It's fine the way it is but maybe that makes sense.